### PR TITLE
Implementing first POM test

### PR DESCRIPTION
**Pull request by manuelq12**
Changes made to implement the POM pattern to execute the tests.

image

image

### CODE REVIEW

File: src/page/index.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export { MenuContentPage } from './menu-content.page';
+export { ProductList } from './product-list.page';
```
Comment (holgiosalos) [inline src/page/index.ts]: please add the Page suffix to all page objects in here. That implies renaming each class in its corresponding file.

File: test/buy-tshirt.spec.ts [hunk 2]
```diff
@@ -1,41 +1,45 @@
...
     await(browser.sleep(3000));
-
-    await $('#form > p > button > span').click();
+    await shippingStep.acceptTermsOfService();
     await(browser.sleep(3000));
-    await $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a').click();
+    await shippingStep.clickCheckOut();
     await(browser.sleep(3000));
-    await $('#cart_navigation > button > span').click();
+    await orderSummary.clickPayByBank();
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: the orderSummaryPage should be the last step, the step to implement here is the paymentStep. I guess you confused the payment step with the orderSummary step.

File: test/buy-tshirt.spec.ts [hunk 3]
```diff
@@ -1,41 +1,45 @@
...
     await(browser.sleep(3000));
-
-    await expect($('#center_column > div > p > strong').getText())
-      .toBe('Your order on My Store is complete.');
+    await bankPayment.clickConfirmOrder();
+    await(browser.sleep(3000));
+    await paymentStep.validateCompleteOrder();
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: The idea of the POM is to model the business behavior of an app component, however, the expectations or validations should remain in the test classes. That being said, the right pattern here should be to have a method to retrieve the orderConfirmation from the summary and compare that confirmation with the expected message on the test class.

File: src/page/index.ts (hunk 1 above)
Comment (manuelq12) [reply src/page/index.ts]: Does all the Classes we make will need to specify the name of the .ts file?

File: test/buy-tshirt.spec.ts (hunk 2 above)
Comment (manuelq12) [reply test/buy-tshirt.spec.ts]: Yes,  for some reason I mistake them. Sorry

File: test/buy-tshirt.spec.ts (hunk 3 above)
Comment (manuelq12) [reply test/buy-tshirt.spec.ts]: Okay, i didn't know that, so is there any other function different from expect, that it must specifically be in the test class and not in the business model classes?

**Review by holgiosalos — APPROVED** [review]

File: test/buy-tshirt.spec.ts (hunk 3 above)
Comment (holgiosalos) [reply test/buy-tshirt.spec.ts]: In general, any function that is used for testing purposes. For example: describe, it, spy, expect, before, after, etc.

**Pull request merged**